snap/hooks/install

```diff
@@ -1,28 +1,45 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# Installation hook for charmed-mongodb snap
 
 set -eux
 
+export CONF="${SNAP_DATA}"/etc/mongod
+export PBM_CONF="${SNAP_DATA}"/etc/pbm
+export DATA="${SNAP_COMMON}"/var/lib/mongodb
+export LOGS="${SNAP_COMMON}"/var/log/mongodb
+export MONGO_CONFIG_FILE="${CONF}"/mongod.conf
+
 # Create the necessary parent directories 
-mkdir -p "${SNAP_DATA}/etc/pbm/"
-mkdir -p "${SNAP_DATA}/etc/mongod/"
-mkdir -p "${SNAP_COMMON}/var/lib/mongodb/"
-mkdir -p "${SNAP_COMMON}/var/log/mongodb/"
+mkdir -p $DATA
+mkdir -p $LOGS
+mkdir -p $PBM_CONF
+mkdir -p $CONF
+
+# If we just created the directory, we set up permissions
+if [ stat -c '%u' "$DATA" == 0 -o ! -d "$DATA" ]; then
+    chmod -R 770 "${SNAP_COMMON}"
+    chmod 750 "${DATA}"
```

Comment:
Should this be recursive?

Reply:
No, as there will be no data in here in case we just created the directory, and we enter this block only if that's the case.
For a bit more exaplanation, there will be a complementary  code on the charm side when attaching a storage to grant the right permissions to the attached storage, like they did in kafka operator.